build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/BwcVersions.java

```diff
@@ -252,6 +252,19 @@ private List<Version> getReleased() {
             .toList();
     }
 
+    public List<Version> getReadOnlyIndexCompatible() {
+        // Lucene can read indices in version N-2
+        int compatibleMajor = currentVersion.getMajor() - 2;
+        return versions.stream()
+            .filter(v -> v.getMajor() == compatibleMajor)
+            .sorted(Comparator.naturalOrder())
+            .toList();
+    }
+
+    public void withLatestReadOnlyIndexCompatible(Consumer<Version> versionAction) {
+        versionAction.accept(getReadOnlyIndexCompatible().getLast());
```

Comment:
Perhaps we should add an explicit assertion that the list is non-empty with a good message?